src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -215,6 +288,87 @@ def store_task_res(self, task_res: TaskRes) -> Optional[UUID]:
         # Return the new task_id
         return task_id
 
+    # pylint: disable=R0911
+    def store_message_res(self, message: Message) -> Optional[UUID]:
+        """Store one Message."""
+        # Validate task
+        errors = validate_message(message, is_reply_message=True)
+        if any(errors):
+            log(ERROR, errors)
+            return None
+
+        res_metadata = message.metadata
+        with self.lock:
+            # Check if the Message it is replying to exists and is valid
+            msg_ins_id = res_metadata.reply_to_message
+            msg_ins = self.message_ins_store.get(UUID(msg_ins_id))
+
+            # Ensure that dst_node_id of original Message matches the src_node_id of
+            # reply Message.
+            if (
+                msg_ins
+                and message
+                and msg_ins.metadata.dst_node_id != res_metadata.src_node_id
+            ):
+                return None
+
+            if msg_ins is None:
+                log(
+                    ERROR,
+                    "Message with `metadata.message_id` %s does not exist.",
+                    msg_ins_id,
+                )
+                return None
+
+            ins_metadata = msg_ins.metadata
+            if ins_metadata.created_at + ins_metadata.ttl <= time.time():
+                log(
+                    ERROR,
+                    "Failed to store Message: the message it is replying to "
+                    "(with id %s) has expired",
+                    msg_ins_id,
+                )
+                return None
+
+            # Fail if the Message TTL exceeds the
+            # expiration time of the Message it replies to.
+            # Condition: ins_metadata.created_at + ins_metadata.ttl ≥
+            #            res_metadata.created_at + res_metadata.ttl
+            # A small tolerance is introduced to account
+            # for floating-point precision issues.
+            max_allowed_ttl = (
+                ins_metadata.created_at + ins_metadata.ttl - res_metadata.created_at
+            )
+            if res_metadata.ttl and (
+                res_metadata.ttl - max_allowed_ttl > MESSAGE_TTL_TOLERANCE
+            ):
+                log(
+                    WARNING,
+                    "Received Message with TTL %.2f exceeding the allowed maximum "
+                    "TTL %.2f.",
+                    res_metadata.ttl,
+                    max_allowed_ttl,
+                )
+                return None
+
+        # Validate run_id
+        if res_metadata.run_id not in self.run_ids:
```

Comment:
I understand that the purpose of this PR is to have a 1:1 copy of existing methods, but maybe we could fix a known bug, right? 
```suggestion
        if res_metadata.run_id != ins_metadata.run_id:
```